### PR TITLE
chore(build): run cucumber tests using NPM scripts instead of Grunt

### PR DESCRIPTION
**Pull request by devoto13**
The main reason for this change is to be able to update Cucumber library as `grunt-cucumberjs` (https://github.com/mavdi/grunt-cucumberjs/issues/110) does not support cucumber@4 and looks pretty abandoned.

Grunt ecosystem is not very active for a while, so we can expect more issues of this sort in the future. Because of this and to keep docs consistent I've also added NPM scripts to run other test types (for now they just proxy to Grunt) and removed Grunt examples from the documentation.

In the next PR I will update `cucumber` and perform some cleanup of the step definitions.

### CODE REVIEW

File: docs/dev/02-making-changes.md [hunk 1]
```diff
@@ -23,29 +23,29 @@ Here are some tips on how to set up a Karma workspace and how to send a good pul
 * Install for development. Use a recent npm version, ignore peerdep warnings
   ```bash
   $ npm install
-  $ rm -rf node_modules/karma
-  $ cd node_modules
-  $ ln -s ../ karma
```
Comment (johnjbarton) [inline docs/dev/02-making-changes.md:28]: I thought this horrible bit was needed by the integration tests.  Linking the just-built new version back into the dependencies allowed the test to run on the just-build version.  So it's not about grunt?

Comment (devoto13) [reply docs/dev/02-making-changes.md:28]: I haven't looked into why exactly it is necessary and how to improve it in this PR. Just noticed that [init script](https://github.com/karma-runner/karma/blob/master/package.json#L499) does all these steps and replaced it with one command.

This seems to be necessary for client tests [at least](https://github.com/karma-runner/karma/blob/master/test/client/karma.conf.js#L49).

Comment (devoto13) [reply docs/dev/02-making-changes.md:28]: @johnjbarton So this is not really necessary for the Cucumber update. Do you want me to remove this change from the PR?

Comment (johnjbarton) [reply docs/dev/02-making-changes.md:28]: That would be better.  I would like to switch to `init` however. `test/client/karma.conf.js` needs a fix too.

Comment (devoto13) [reply docs/dev/02-making-changes.md:28]: @johnjbarton Okey, will revert this lines and submit as a separate PR instead.

**Pull request merged**